### PR TITLE
Enhanced satori client

### PR DESCRIPTION
**Pull request by fsufyan**
Several fixes and api enhancements, also added more tests.

### CODE REVIEW

File: Sources/Satori/Models/Experiment.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// All experiments that this identity is involved with.
+public struct ExperimentList {
+    /// All experiments for this identity.
+    let experiments: [ApiExperiment]
```
Comment (lugehorsam) [inline Sources/Satori/Models/Experiment.swift:20]: Should this be `let experiments: [Experiment]`?

File: Sources/Satori/Models/Flag.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// All flags available to the identity
+public struct FlagList {
+    /// All flags
+    let flags: [ApiFlag]
```
Comment (lugehorsam) [inline Sources/Satori/Models/Flag.swift:20]: Should this be `[Flag]`

**Review by lugehorsam — CHANGES_REQUESTED** [review]
just two minor comments

**Comment by fsufyan** [comment]
> just two minor comments

Fixed, the type in Experiment and Flag lists is left without optional to avoid optional chaining ?.
Users could check for empty list instead of nil.

**Review by lugehorsam — APPROVED** [review]

**Pull request merged**